pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -381,6 +349,40 @@ func (s *FlinkStateMachine) handleApplicationSavepointing(ctx context.Context, a
 	return statusUnchanged, nil
 }
 
+func (s *FlinkStateMachine) handleApplicationRecovering(ctx context.Context, app *v1beta1.FlinkApplication) (bool, error) {
+	// we're in the middle of a deploy, and savepointing has failed in some way... we're going to try to recover
+	// and push through if possible
+	if rollback, reason := s.shouldRollback(ctx, app); rollback {
+		// we failed to recover, attempt to rollback
+		s.flinkController.LogEvent(ctx, app, corev1.EventTypeWarning, "JobsSubmissionFailed",
```

Comment:
Should this event be something other than `JobsSubmissionFailed`?

Reply:
Yes, good call; updated to "RecoveryFailed"